src/frontend/planner_test/tests/testdata/batch_dist_agg.yaml

```diff
@@ -160,9 +159,10 @@
     BatchExchange { order: [], dist: Single }
     └─BatchProject { exprs: [max(s.v)] }
       └─BatchHashAgg { group_key: [s.k], aggs: [max(s.v)] }
-        └─BatchScan { table: s, columns: [s.k, s.v], distribution: UpstreamHashShard() }
+        └─BatchExchange { order: [], dist: HashShard(s.k) }
+          └─BatchScan { table: s, columns: [s.k, s.v], distribution: Single }
   batch_local_plan: |
-    BatchProject { exprs: [max(s.v)] }
-    └─BatchHashAgg { group_key: [s.k], aggs: [max(s.v)] }
-      └─BatchExchange { order: [], dist: Single }
-        └─BatchScan { table: s, columns: [s.k, s.v], distribution: UpstreamHashShard() }
+    BatchExchange { order: [], dist: Single }
+    └─BatchProject { exprs: [max(s.v)] }
+      └─BatchHashAgg { group_key: [s.k], aggs: [max(s.v)] }
+        └─BatchScan { table: s, columns: [s.k, s.v], distribution: Single }
```

Comment:
Take this for example, root fragment task is always executed on FE, while one is exchange singleton and another is hash agg. 

What's the benefit of doing so? 🤔

Reply:
It's for this bug: https://github.com/risingwavelabs/risingwave/issues/4164